functests/3_performance_status/status.go

```diff
@@ -133,4 +136,66 @@ var _ = Describe("Status testing of performance profile", func() {
 			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		})
 	})
+
+	Context("Status reports degraded condition", func() {
+		It("Should report Degraded status if overlapping cpus are configured", func() {
+			if discovery.Enabled() {
+				Skip("Discovery mode enabled, test skipped because it creates incorrect profiles")
+			}
+
+			newRole := "worker-overlapping"
+			newLabel := fmt.Sprintf("%s/%s", testutils.LabelRole, newRole)
+
+			reserved := performancev1.CPUSet("0-3")
+			isolated := performancev1.CPUSet("0-7")
+
+			overlappingProfile := &performancev1.PerformanceProfile{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "PerformanceProfile",
+					APIVersion: performancev1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "profile-overlapping-cpus",
+				},
+				Spec: performancev1.PerformanceProfileSpec{
+					CPU: &performancev1.CPU{
+						Reserved: &reserved,
+						Isolated: &isolated,
+					},
+					NodeSelector: map[string]string{newLabel: ""},
+					RealTimeKernel: &performancev1.RealTimeKernel{
+						Enabled: pointer.BoolPtr(true),
+					},
+					NUMA: &performancev1.NUMA{
+						TopologyPolicy: pointer.StringPtr("restricted"),
+					},
+				},
+			}
+			err := testclient.Client.Create(context.TODO(), overlappingProfile)
+			Expect(err).ToNot(HaveOccurred(), "error creating overlappingProfile: %v", err)
+			defer func() {
+				Expect(testclient.Client.Delete(context.TODO(), overlappingProfile)).ToNot(HaveOccurred())
+				Expect(profiles.WaitForDeletion(overlappingProfile, 60*time.Second)).ToNot(HaveOccurred())
+
+				Consistently(func() corev1.ConditionStatus {
+					return mcps.GetConditionStatus(testutils.RoleWorkerCNF, machineconfigv1.MachineConfigPoolUpdating)
+				}, 30, 5).Should(Equal(corev1.ConditionFalse), "Machine Config Pool is updating, and it should not")
+
+			}()
+
+			nodeLabels := map[string]string{
+				newLabel: "",
+			}
+
+			var cond *v1.Condition
+			Eventually(func() bool {
+				cond = profiles.GetCondition(nodeLabels, v1.ConditionDegraded)
```

Comment:
I had a hard time understanding what this does. Would it be possible to have a local "isProfileDegraded" function (or something along that line) wrapping the code?